### PR TITLE
Feature/PATRO23-373 Share budget improvment

### PR DESCRIPTION
**Pull request by IzabelaPalubicka**
https://tracker.intive.com/jira/browse/PATRO23-373

### CODE REVIEW

**Review by Kepeczsz — APPROVED** [review]

File: src/modules/budget/application/UserBudgets/DeleteUserBudget/DeleteUserBudgetList.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			.Select(x => x.Id)
+			.ToArrayAsync(cancellationToken: cancellationToken);
+
+		var userBudgetsList = await this.userBudgetRepository.GetByIds(userBudgetIds);
+
+		foreach (var userBudget in userBudgetsList)
+		{
+			userBudget.Delete();
+
+			await this.userBudgetRepository.RemovePersist(userBudget);
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/UserBudgets/DeleteUserBudget/DeleteUserBudgetList.cs:49]: Remove should be enough.

File: src/modules/budget/application/UserBudgets/DeleteUserBudget/DeleteUserBudgetList.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	public HandleDeleteUserBudget(BudgetDbContext budgetDbContext, IRepository<UserBudgetAggregate, Guid> userBudgetRepository)
+	{
+		this.budgetDbContext = budgetDbContext;
+		this.userBudgetRepository = userBudgetRepository;
+	}
+
+	/// <inheritdoc/>
+	public async Task Handle(DeleteUserBudgetList command, CancellationToken cancellationToken)
+	{
+		var userBudgetIds = await this.budgetDbContext.UserBudget
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/UserBudgets/DeleteUserBudget/DeleteUserBudgetList.cs:38]: I'm not sure about using DbContext in commands We should use queries and repositories only.

File: src/modules/budget/application/UserBudgets/ShareBudget/ShareBudget.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	private readonly ICommandBus commandBus;
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="HandleShareBudget"/> class.
+	/// </summary>
+	/// <param name="budgetDbContext">Object representing the database context used to retrieve data.</param>
+	/// <param name="commandBus">Command bus.</param>
+	public HandleShareBudget(BudgetDbContext budgetDbContext, ICommandBus commandBus)
+	{
+		this.budgetDbContext = budgetDbContext;
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/UserBudgets/ShareBudget/ShareBudget.cs:34]: I'm not sure about using DbContext in commands We should use queries and repositories only.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/modules/budget/application/UserBudgets/DeleteUserBudget/DeleteUserBudgetList.cs (hunk 2 above)
Comment (mateuszgiedzinski) [reply src/modules/budget/application/UserBudgets/DeleteUserBudget/DeleteUserBudgetList.cs:38]: I confirmed with @mateuszmuszynski  that this should come from query handler

File: src/modules/budget/application/UserBudgets/AddingUserBudget/AddUserBudgetListValidator.cs [hunk 4]
```diff
@@ -35,9 +35,10 @@ public AddUsersToBudgetValidator(IKeycloakService keycloakService, BudgetDbConte
 			.WithMessage("{PropertyName}: Budget with id {PropertyValue} does not exist.").WithErrorCode("1.11");
 
 		this.RuleFor(x => x.UsersIds)
-			.NotEmpty()
 			.NotNull()
-			.CustomAsync(this.AreExistingUsersIds);
+			.Custom(this.IsUserIdDupliacted)
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/UserBudgets/AddingUserBudget/AddUserBudgetListValidator.cs:39]: please assign some error code for your validation rule.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/modules/budget/application/UserBudgets/GettingUserBudget/GetUserBudgetListValidator.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	/// Initializes a new instance of the <see cref="GetUserBudgetListValidator"/> class.
+	/// </summary>
+	/// <param name="budgetDbContext">Budget repository to validate budget ids.</param>
+	public GetUserBudgetListValidator(BudgetDbContext budgetDbContext)
+	{
+		this.budgetDbContext = budgetDbContext;
+
+		this.RuleFor(x => x.BudgetId)
+			.NotEmpty()
+			.NotNull()
```
Comment (pzajaczkowski) [inline src/modules/budget/application/UserBudgets/GettingUserBudget/GetUserBudgetListValidator.cs:25]: `.NotEmpty()` covers null case, not null not needed

**Review by pzajaczkowski — APPROVED** [review]

**Review by mateuszgiedzinski — APPROVED** [review]

**Pull request merged**